webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/RemoteSourceDescriptorService.scala

```diff
@@ -59,63 +62,43 @@ class RemoteSourceDescriptorService @Inject()(dSRemoteWebknossosClient: DSRemote
       magLocator: MagLocator)(implicit ec: ExecutionContext): Fox[RemoteSourceDescriptor] =
     for {
       credentialBox <- credentialFor(magLocator: MagLocator).shiftBox
-      uri <- uriForMagLocator(baseDir, datasetId, layerName, magLocator).toFox
-      remoteSource = RemoteSourceDescriptor(uri, credentialBox.toOption)
+      uri <- resolveMagPath(baseDir, datasetId, layerName, magLocator).toFox
+      remoteSource = RemoteSourceDescriptor(UPath.fromStringUnsafe(uri.toString), credentialBox.toOption)
     } yield remoteSource
 
-  def uriFromPathLiteral(pathLiteral: String, localDatasetDir: Path, layerName: String): URI = {
-    val uri = new URI(pathLiteral)
-    if (DataVaultService.isRemoteScheme(uri.getScheme)) {
-      uri
-    } else if (uri.getScheme == null || uri.getScheme == DataVaultService.schemeFile) {
-      val localPath = Path.of(uri.getPath)
-      if (localPath.isAbsolute) {
-        if (localPath.toString.startsWith(localDatasetDir.getParent.toAbsolutePath.toString) || dataStoreConfig.Datastore.localDirectoryWhitelist
-              .exists(whitelistEntry => localPath.toString.startsWith(whitelistEntry)))
-          uri
-        else
-          throw new Exception(
-            s"Absolute path $localPath in local file system is not in path whitelist. Consider adding it to datastore.localDirectoryWhitelist")
-      } else { // relative local path, resolve in dataset dir
-        val pathRelativeToDataset = localDatasetDir.resolve(localPath).normalize
-        val pathRelativeToLayer = localDatasetDir.resolve(layerName).resolve(localPath).normalize
-        if (pathRelativeToDataset.toFile.exists) {
-          pathRelativeToDataset.toUri
-        } else {
-          pathRelativeToLayer.toUri
-        }
-      }
-    } else {
-      throw new Exception(s"Unsupported path: $localDatasetDir")
-    }
-  }
-
-  def resolveMagPath(datasetDir: Path, layerDir: Path, layerName: String, magLocator: MagLocator): URI =
+  def resolveMagPath(localDatasetDir: Path, layerDir: Path, layerName: String, magLocator: MagLocator): UPath =
```

Comment:
Just a rather general question: Most of the method you are working with `Path`s and not `UPath`s while the  `UPath` with its / operation seems more convinient. So that might be a good thing to change :thinking:. But not strong opinion here as the `LocalUPath` uses a `Path` under the hood anyway